numpy/core/src/multiarray/descriptor.c

```diff
@@ -251,7 +251,7 @@ static PyArray_Descr *
 _convert_from_tuple(PyObject *obj, int align)
 {
     if (PyTuple_GET_SIZE(obj) != 2) {
-        _report_generic_error();
+        PyErr_Format(PyExc_TypeError, "PyTuple must have size 2, got [%d]", PyTuple_GET_SIZE(obj));
```

Comment:
Best to use the python names (tuple, object) rather than the CPython names (PyTuple, PyObject). Same throughout.

Reply:
Agreed, will change in applicable places.